src/net.cpp

```diff
@@ -1308,19 +1316,20 @@ void CConnman::SocketHandlerConnected(const std::vector<CNode*>& nodes,
             {
                 bool notify = false;
 
-                // a map to store the bytes by message type
-                mapMsgTypeSize msgtype_bytes;
+                // a map to store each message type mapped to the total message count,
+                // as well as the total number of message bytes
+                std::map<std::string, std::tuple<int, uint64_t>> msgtype_countbytes;
 
                 // initialize all message type values to zero
                 for (const std::string& msg : getAllNetMessageTypes())
```

Comment:
ah, I figured this out (also captured in the send & receive diagram) - `ThreadSocketHandler` calls `SocketHandler` on loop, which goes through each node. so in the end, we wouldn't want to initialize this data structure here every time because of the resource usage. 

one idea is to populate the stats when pulling the data off `vRecvMsg` later in this function, instead of when we are populating the member in `ReceiveMsgBytes`. I believe this should allow us to remove the temporary variable `msgtype_countbytes`.

Reply:
Since this comment, I've made updates to hold off on populating stats until later in the function, when the data is pulled off vRecvMsg. 

The reason for `msgtype_countbytes` is that `vRecvMsg` can have multiple messages of the same type. I agree that `msgtype_countbytes` is a larger data structure that I don't want to keep initializing there's another way. The only alternative I see is to call `RecordMsgStatsRecv()` each time a message is processed off of `vRecvMsg` in the loop. 

My gut feeling is this takes less resources but want to get your thoughts on it first.

Reply:
hm, I don't remember enough context to evaluate off hand, but will keep this in mind to evaluate when I'm reviewing PR#4 :)

Reply:
I'm going to follow up on this in response to Vasil's comment below :D https://github.com/satsie/bitcoin/pull/3/files#r1104437828 Closing this one!